### ISSUE TITLE
Regex doesn't work for ignoreDependencies

### ISSUE THREAD

**Issue by jakebailey** [webpro/knip#392]
I tried to use a regex to ignore multiple unused dependencies, but nothing I tried seemed to work. Our config is here: https://github.com/microsoft/DefinitelyTyped-tools/blob/main/.knip.jsonc#L64

I tried `typescript-*`, `typescript-.*`, and more, but none worked; only manually writing them out did.

Additionally, using a `js` config for this and passing in a regex object broke at runtime.

(We are still on v2 officially, but I tested v3 and the behavior was the same.)

**Comment by webpro** [comment]
I'll check it out on that repo tomorrow.

Sure you tried the latest? This should be covered: https://github.com/webpro/knip/blob/main/packages/knip/fixtures/ignore-dependencies-binaries/knip.ts and it creates a regex from strings (with special chars) in JSON too.

**Comment by jakebailey** [comment]
https://github.com/microsoft/DefinitelyTyped-tools/pull/880

I guess it has to be an executable config? The docs say otherwise.

**Comment by jakebailey** [comment]
Okay, it kinda works with a `.js` config and a RegExp object, but it complains that it's unused, even though it isn't:

```
Configuration issues (1)
Unused item in ignoreDependencies (workspace: packages/typescript-packages): /typescript-.*/
✂️  Excellent, Knip found no issues.
```

Removing the config:

```
Unused dependencies (9)
typescript-4.6  packages/typescript-packages/package.json
typescript-4.7  packages/typescript-packages/package.json
typescript-4.8  packages/typescript-packages/package.json
typescript-4.9  packages/typescript-packages/package.json
typescript-5.0  packages/typescript-packages/package.json
typescript-5.1  packages/typescript-packages/package.json
typescript-5.2  packages/typescript-packages/package.json
typescript-5.3  packages/typescript-packages/package.json
typescript-5.4  packages/typescript-packages/package.json
```

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v3.8.1. See [Release 3.8.1](https://github.com/webpro/knip/releases/tag/3.8.1) for release notes._

**Comment by webpro** [comment]
Fixed it in v3.8.1 (the mapping of strings to regexes did not apply to non-root workspaces).

The warning is from `@definitelytyped/typescript-versions` which is listed in `dependencies`, but also referenced in the source code. You can adjust the pattern to e.g. `"^typescript-.*"` to help with that. Knip could be more helpful with pointing that out, though.

**Comment by jakebailey** [comment]
Thanks, that all worked!

**Comment by webpro** [comment]
Excellent, thanks for the feedback

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v4.0.0. See [Release 4.0.0](https://github.com/webpro/knip/releases/tag/4.0.0) for release notes._
